### DISCUSSION TITLE
fail to compile `diesel = "2.1.1"`

### DISCUSSION THREAD

**Discussion by JasonLi-cn** [diesel-rs/diesel#3763] [Q&A]
## Setup
[dependencies]
diesel = "2.1.1"

### Versions

- **Rust:** nightly-2023-05-15-x86_64-unknown-linux-gnu
- **Diesel:** 2.1.1
- **Database:** pg
- **Operating System** ubuntu 18.04

### Feature Flags

- **diesel:** `["uuid", "postgres", "i-implement-a-third-party-backend-and-opt-into-breaking-changes", "network-address", "chrono", "numeric"]`

### What is the actual output?

```shell
158 |         .load_stream(conn)
    |          ^^^^^^^^^^^ the trait `CompatibleType<_, diesel::pg::Pg>` is not implemented for `diesel::expression::select_by::SelectBy<MyColumn, diesel::pg::Pg>`

158 |         .load_stream(conn)
    |          ----------- ^^^^ the trait `CompatibleType<_, diesel::pg::Pg>` is not implemented for `diesel::expression::select_by::SelectBy<MyColumn, diesel::pg::Pg>`
```

## Checklist

- [ ] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) and the [discussion forum](https://github.com/diesel-rs/diesel/discussions) for similar possible closed issues.
 
- [ ] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)
- [ ] This issue can be reproduced without requiring a third party crate

**Comment by weiznich** [comment]
The error message indicates that this seems to be a problem in your code, therefore I've converted this to a support discussion. It's unfortunately not possible to help you without more information. You might want to include the following information:

* The relevant code that causes the error, including the schema.rs file, all relevant type defs not coming from diesel/rust-std and any other relevant part of your code that is required to reproduce the problem
* The exact error message as emitted by rustc
